File: 2014/FASE 2/JAVASCRIPT/q-erradas-js/voo.js
function calcularDuracaoEFusoHorario(idaPartida, voltaChegada) {
    // Extrair as horas e minutos dos horários de ida e volta
    const [idaHora, idaMinuto] = idaPartida.split(':').map(Number);
    const [voltaHora, voltaMinuto] = voltaChegada.split(':').map(Number);

    // Calcular a duração do voo em minutos
    const duracaoVoo = (voltaHora * 60 + voltaMinuto) - (idaHora * 60 + idaMinuto);

    // Determinar a diferença de fusos horários
    let diferencaFusos = voltaHora - idaHora;

    // Ajustar a diferença de fusos horários se ultrapassar os limites (-12 < delta ≤ 12)
    if (diferencaFusos > 12) {
        diferencaFusos -= 24;
    } else if (diferencaFusos <= -12) {
        diferencaFusos += 24;
    }

    return [duracaoVoo, diferencaFusos];
}

// Função para converter a diferença de fusos horários para formato de texto
function formatarDiferencaFusos(diferencaFusos) {
    if (diferencaFusos > 0) {
        return `${diferencaFusos}`;
    } else if (diferencaFusos < 0) {
        return `-${Math.abs(diferencaFusos)}`;
    } else {
        return '0';
    }
}

// Leitura da entrada
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('', (entrada) => {
    const [idaPartida, voltaChegada, voltaPartida, idaChegada] = entrada.split(' ');

    // Calcular a duração do voo e a diferença de fusos horários
    const [duracaoVoo, diferencaFusos] = calcularDuracaoEFusoHorario(idaPartida, voltaChegada);

    // Imprimir a duração do voo em minutos e a diferença entre os fusos horários
    console.log(`${duracaoVoo} ${formatarDiferencaFusos(diferencaFusos)}`);

    rl.close();
});

File: 2014/FASE 2/PYTHON/q-erradas-py/voo.py
def converter_horario_para_minutos(horario):
    # Converter o horário para minutos desde a meia-noite
    horas, minutos = map(int, horario.split(':'))
    return horas * 60 + minutos

def calcular_duracao_voo(p_A, c_B, p_B, c_A):
    # Converter horários para minutos
    partida_A = converter_horario_para_minutos(p_A)
    chegada_B = converter_horario_para_minutos(c_B)
    partida_B = converter_horario_para_minutos(p_B)
    chegada_A = converter_horario_para_minutos(c_A)

    # Calcular a duração do voo
    duracao_voo = chegada_B - partida_A

    # Calcular a diferença de fusos horários entre as cidades A e B
    diferenca_fusos = (partida_B - chegada_A) // 60

    # Determinar se B está à frente ou atrás de A em termos de fusos horários
    if chegada_B < partida_A:
        diferenca_fusos -= 1

    return duracao_voo, diferenca_fusos

def main():
    # Ler os horários dos voos
    p_A, c_B, p_B, c_A = input().split()

    # Calcular a duração do voo e a diferença de fusos horários
    duracao_voo, diferenca_fusos = calcular_duracao_voo(p_A, c_B, p_B, c_A)

    # Imprimir resultado
    print(duracao_voo, diferenca_fusos)

if __name__ == "__main__":
    main()
